### ISSUE TITLE
Convert to tflite

### ISSUE THREAD

**Issue by Sunnycheey** [yeephycho/tensorflow-face-detection#54]
I try to convert **.pb** file to **.tflite** file so that I can deploy the model to android. 

But when I try to use **_tflite_convert_** to do the convert, it failed.  Then, I follow the instruction [here](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/running_on_mobile_tensorflowlite.md) which indicates that **ckpt** file is needed to generate **.pb** and then **.tflite**(But in this repo only pb file was offered).

I also try to use the code [here](https://github.com/yeephycho/tensorflow-face-detection/issues/42) to do the 'pb to ckpt' transform, but when I try to use the generated ckpt file to genrate .pb file, I failed again.

So can you **offer** the ckpt file, or can you indicate how to do the wider face dataset clean to train the model and get the ckpt file.

Thanks.

**Comment by Sunnycheey** [comment]
Fixed by training my own model.

**Comment by fariagu** [comment]
@Sunnycheey I'm having the same problem you had, but I'm more used to working with keras, so the tflite_convert is also not working. Could you please explain how you generated the tflite file, or maybe even share the file itself? Thanks!

**Comment by Sunnycheey** [comment]
> @Sunnycheey I'm having the same problem you had, but I'm more used to working with keras, so the tflite_convert is also not working. Could you please explain how you generated the tflite file, or maybe even share the file itself? Thanks!

Have you solve your problem?

I can share my train process with you simply.

Firstly, you need write a small script to convert wider face database to tfrecord format, then you can train the model and get your own ckpt file. You can learn how to generate tfrecord in [here](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/using_your_own_dataset.md) and you can find pre-trained model in [here](https://github.com/tensorflow/models/blob/master/research/object_detection/g3doc/detection_model_zoo.md)

**Comment by fariagu** [comment]
Yes, I did fix it. Sorry for not updating.

I didn't train my own model, I followed the method described in the discussion of this issue (https://github.com/yeephycho/tensorflow-face-detection/issues/42).

Thanks for replying though 😄

**Comment by Sunnycheey** [comment]
To be honest, I try the solution decribed in the issue ([#42](https://github.com/yeephycho/tensorflow-face-detection/issues/42)), but I can't figure it out, so I train my own model. The problem is the test result of my own trained model is not as good as this repo. Have you finished the convert? Can you offered the tflite file itself?

Thanks.

**Comment by fariagu** [comment]
Sure thing. I can't say for sure how good the accuracy is but running on android the results are pretty decent.

The generated file is not quantized, though. Still haven't figured that part out.
[detect.zip](https://github.com/yeephycho/tensorflow-face-detection/files/3181846/detect.zip)

**Comment by Sunnycheey** [comment]
Well the generated file of my own trained model is quantized. Maybe floating point detect file can achieve more accuracy result but slower running speed. If you'd like to train your model, I can offer you some help. By the way, can you send the zip file directly to my email (lihuichaolihc@gmail.com)? since downloading file on github is so slow in CN...

**Comment by Sunnycheey** [comment]
I use your float point model to build tensorflow demo android app. The app build successfully, but it flash back. I use logcat command and see the error message _‘Cannot convert between a TensorFlowLite buffer with 3145728 bytes and a ByteBuffer with 1080000 bytes.’_. In issue #42, I found you set the input tensor size as 512*512*3, but I set input tensor size as 300*300*3 in my tflite file which worked in my android phone. Have you come across this problem? if yes, what's the solution?

Thanks.

**Comment by fariagu** [comment]
Yes I think I can help you solve that, but maybe you should post this
question as a comment on that github issue. That way if someone has the
same problem it can stay there as a solution.

But long story short you have to change the default input size. The demo
model uses a picture of 300x300x3 and the face detection model expects an
input of 512x512x3

**Comment by Sunnycheey** [comment]
Sorry for deleting the comment. By change _TF_OD_API_INPUT_SIZE_ variable in [there](https://github.com/tensorflow/tensorflow/blob/ac6e97967698a9de32a04d48a4ef4b355f8c6e65/tensorflow/examples/android/src/org/tensorflow/demo/DetectorActivity.java) from defalut 300 to 512, I solved the problem. I delete the comment since I think the problem is somewhat stupid to ask. The variable name explicitly tell me how to set the input tensor size.

Thanks anyway.

**Comment by fariagu** [comment]
It's not a stupid question, I had a similar problem but I had noone to ask
for help. There might be someone in the future who would appreciate having
their "stupid" question already be asked by someone else
